crates/gen/src/types/function.rs

```diff
@@ -34,6 +34,18 @@ impl Function {
             link = "onecoreuap";
         }
 
+        let static_lib = def
+            .attributes()
+            .filter_map(|attribute| match attribute.name() {
+                "StaticLibraryAttribute" => Some(attribute.args()[0].1.gen_value()),
+                _ => None,
+            })
+            .next();
+        let link_attr = match static_lib {
```

Comment:
Can we imagine a scenario where a developer actually wants dynamic loading even though a static lib is available?

Reply:
I can't think of any reason to do that, other than maybe avoiding transitive linker dependencies (see below). Right now it's up to the developer/implementer who builds the winmd. As long as the win32metadata SDK works well enough to build a custom winmd, a user of the Windows crate who prefers to use DLLs can build their own without the `--staticLibs` flag/`<StaticLibrary />` prop in the winmd project. A project that provides both binaries (like WebView2) could even provide a static library version and a DLL version of the winmd.

Do you prefer adding an option somewhere in `windows-rs` to suppress this downstream?

Reply:
No, I'd prefer not to add an option for this. Just making sure we've at least considered this.

Reply:
Pros and cons for sure. What's the suggested change?

Reply:
I'd leave it the way it is - prefer lib if available - other languages (that don't have a linker) ignore. Component authors that wish to offer both could always include two function definitions.